test/certUtils.test.js

```diff
@@ -117,14 +117,14 @@ describe("cert utils", function() {
 			it("returns correct serial for attestation", function() {
 				const cert = new Certificate(h.certs.yubiKeyAttestation);
 				const serial = cert.getSerial();
-				assert.strictEqual(serial, "Yubico U2F EE Serial 1432534688");
+				assert.strictEqual(serial, "1432534688");
```

Comment:
Hmm this could actually be an issue - it's likely that some users are relying on the format to not change and contain the extended information. This means that we will need a migration path, where new users get only the serial while old users can maintain the format. This is commonly done by introducing a new method and giving out deprecation warnings, or allowing to use the old format with some optional parameter.  Do you have a preferred solution or idea?

Reply:
That's a great point, thank you! Well, the previous behavior of `getSerial` is the same as that of `getCommonName`. What if I were to do something like this? I think it could also be a good idea to add another test for when the method is called with `returnSerialNumber = true`. Otherwise, I'm open to other suggestions!
```
getSerial(returnSerialNumber = false) {
    return (returnSerialNumber) ? this._cert.serialNumber.valueBlock.toString() : this.getCommonName();
}
```

Reply:
That'd be a good idea, however I'm not sure if the naming is clear enough for people - they might be confused which one they are expected to use.

What do you think about having a parameter like "compatibility" that is an optional parameter? 

- If it is not defined, we print a deprecation notice and ask users to provide a parameter to the method. It then automatically uses "v1", but does not print the additional deprecation warning below.
- If "v1" is used, we print a deprecation notice and ask the user to convert to "v2", while returning the commonName. 
- If the parameter is "v2", we return the serial.

In such a case, we need to update our tests and documentation as well to always use "v2". Given enough time, we could use the same mechanism to update the method again or remove v1 in the future.

Reply:
That sounds good! Thanks for your patience and explanations. You're teaching me a lot here :)

Reply:
Ok I added the "compatibility" parameter which prints a deprecation notice if it's not set to "v2". I also updated the tests to use "v2". Please let me know if that's what you had in mind or if you have any other suggestions!